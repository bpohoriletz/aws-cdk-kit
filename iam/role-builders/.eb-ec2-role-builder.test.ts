import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import EbEc2RoleBuilder from './eb-ec2-role-builder';

describe('new EbEc2RoleBuilder()', () => {
  let instance: EbEc2RoleBuilder;
  let stack: cdk.Stack;

  beforeEach(() => {
    stack = new cdk.Stack();
    instance = new EbEc2RoleBuilder(stack, 'EbEc2Role');
  });

  test('construct EC2 role for ElasticBeanstalk', () => {
    instance.setManagedPolicies().setInlinePolicies(['*']);

    expect(Template.fromStack(stack)).toMatchSnapshot();
  });
});
