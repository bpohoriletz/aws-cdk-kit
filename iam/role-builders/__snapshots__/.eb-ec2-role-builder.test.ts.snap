// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new EbEc2RoleBuilder() construct EC2 role for ElasticBeanstalk 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "EbEc2Role46292FC8": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore",
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "allowEc2PutStatisticsD9C1A435": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "elasticbeanstalk:PutInstanceStatistics",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "allowEc2PutStatisticsD9C1A435",
        "Roles": [
          {
            "Ref": "EbEc2Role46292FC8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "allowEc2UploadLogsA49BD994": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "allowEc2UploadLogsA49BD994",
        "Roles": [
          {
            "Ref": "EbEc2Role46292FC8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "allowEc2UploadLogsToEbRegionalBucket40230F17": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:ListBucket",
                "s3:ListBucketVersions",
                "s3:GetObject",
                "s3:GetObjectVersion",
                "s3:PutObject",
              ],
              "Effect": "Allow",
              "Resource": [
                "*",
                "*/resources/*",
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "allowEc2UploadLogsToEbRegionalBucket40230F17",
        "Roles": [
          {
            "Ref": "EbEc2Role46292FC8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
